source/adapters/level_zero/event.cpp

```diff
@@ -964,8 +966,7 @@ ur_result_t CleanupCompletedEvent(ur_event_handle_t Event, bool QueueLocked,
 ur_result_t EventCreate(ur_context_handle_t Context, ur_queue_handle_t Queue,
                         bool HostVisible, ur_event_handle_t *RetEvent) {
 
-  bool ProfilingEnabled =
-      !Queue || (Queue->Properties & UR_QUEUE_FLAG_PROFILING_ENABLE) != 0;
+  bool ProfilingEnabled = !Queue || Queue->isProfilingEnabled();
```

Comment:
I know you did not change this, but it looks weird that when Queue == null then we create an event from a profiled pool

Reply:
@smaslov-intel : I know, I saw that and it looks weird. Maybe that was added as a WA, or a better logic could be used. i'm planning on addressing that in another patch.